python/tank/context.py

```diff
@@ -793,26 +784,17 @@ def to_dict(self):
 
         :returns: A dictionary representing the context.
         """
-        return self._to_dict()
-
-    def _to_dict(self, keep_name=True):
-        """
-        See documentation from :meth:`to_dict`.
-
-        :param bool keep_name: When set to ``True``, the name will be kept inside the
-            entities. Defaults to ``True``.
-        """
         return {
-            "project": self._cleanup_entity(self.project, keep_name),
-            "entity": self._cleanup_entity(self.entity, keep_name),
-            "user": self._cleanup_entity(self.user, keep_name),
-            "step": self._cleanup_entity(self.step, keep_name),
-            "task": self._cleanup_entity(self.task, keep_name),
-            "additional_entities": [self._cleanup_entity(entity, keep_name) for entity in self.additional_entities],
-            "source_entity": self._cleanup_entity(self.source_entity, keep_name)
+            "project": self._cleanup_entity(self.project),
+            "entity": self._cleanup_entity(self.entity),
+            "user": self._cleanup_entity(self.user),
+            "step": self._cleanup_entity(self.step),
+            "task": self._cleanup_entity(self.task),
+            "additional_entities": [self._cleanup_entity(entity) for entity in self.additional_entities],
+            "source_entity": self._cleanup_entity(self.source_entity)
         }
 
-    def _cleanup_entity(self, entity, keep_name):
+    def _cleanup_entity(self, entity):
```

Comment:
This change was caused by the revert of my original pickle fix that was merged some time ago in the Python 3 branch.